lib/src/aura_internal_wallet_ipml.dart

```diff
@@ -0,0 +1,168 @@
+  }) async {
+    try {
+      // Generate a random mnemonic with a strength of 256 bits.
+      final List<String> mnemonic = Bip39.generateMnemonic(strength: 256);
+
+      // Derive a wallet from the generated mnemonic.
+      final Wallet wallet = Wallet.derive(mnemonic, Storehouse.networkInfo);
+
+      // Create and return a ComprehensiveWallet instance with the derived wallet details.
+      return ComprehensiveWallet(
+        auraWallet: AuraWalletImpl(
+          walletName: walletName,
+          bech32Address: wallet.bech32Address,
+          environment: Storehouse.environment,
+        ),
+        mnemonic: mnemonic.join(' '),
+        privateKey: HEX.encode(wallet.privateKey),
+      );
+    } catch (e) {
+      // Handle any exceptions that might occur during wallet creation.
+      throw AuraInternalError(
+        ErrorCode.WalletCreationError,
+        'Error creating a random HD wallet: $e',
+      );
+    }
+  }
+
+  // Restore an HD wallet using a provided passphrase.
+  @override
+  Future<AuraWallet> restoreHDWallet({
+    required String passPhrase,
+    String walletName = CONST_DEFAULT_WALLET_NAME,
+  }) async {
+    try {
+      // Check if the provided passphrase is a valid mnemonic.
+      bool isMnemonic = AuraWalletHelper.checkMnemonic(mnemonic: passPhrase);
+
+      // If it's a valid mnemonic, throw an error.
+      if (!isMnemonic) {
+        throw AuraInternalError(
+          ErrorCode.InvalidPassphrase,
+          'Invalid passphrase provided.',
+        );
+      }
+
+      // Derive a wallet from the provided passphrase.
+      final Wallet wallet =
+          Wallet.derive(passPhrase.split(' '), Storehouse.networkInfo);
+
+      // Save the wallet details to storage.
+      await Storehouse.storage.saveWalletToStorage(
+        walletName: walletName,
+        passphrase: passPhrase,
+        walletAddress: wallet.bech32Address,
+      );
+
+      // Create and return an AuraWalletImpl instance with the restored wallet details.
+      return AuraWalletImpl(
+        walletName: walletName,
+        bech32Address: wallet.bech32Address,
+        environment: Storehouse.environment,
+      );
+    } catch (e) {
+      // Handle any exceptions that might occur during wallet restoration.
+      throw AuraInternalError(
+        ErrorCode.WalletRestorationError,
+        'Error restoring HD wallet: $e',
+      );
+    }
+  }
+
+  // Load a stored wallet using its walletName.
+  @override
+  Future<AuraWallet?> loadStoredWallet({
+    String walletName = CONST_DEFAULT_WALLET_NAME,
+  }) async {
+    try {
+      // Attempt to read the passphrase from storage.
+      String? passPhrase =
+          await Storehouse.storage.readWalletPassPhrase(walletName: walletName);
+
+      // If the passphrase is null, return null (no wallet found).
+      if (passPhrase == null) {
+        return null;
+      }
+
+      // Derive a wallet from the stored passphrase.
+      final Wallet wallet =
```

Comment:
Should call to `restoreHDWallet` instead of copy/paste code

Reply:
Because , they don't have the same rules or logic. The "loadStoreWallet" method get wallet from secure storage. It will check user verified two times. The "restoreHDWallet" method must to save pass phrase  to secure store and it will check user verified one more time. So , I think we should not call to "restoreHDWallet" method

Reply:
Check